python/deltalake/table.py

```diff
@@ -459,6 +459,30 @@ def pyarrow_schema(self) -> pyarrow.Schema:
         )
         return self.schema().to_pyarrow()
 
+    def restore(
+        self,
+        version_to_restore: Optional[int] = None,
+        datetime_to_restore: Optional[str] = None,
+        ignore_missing_files: bool = False,
+        protocol_downgrade_allowed: bool = False,
+    ) -> Dict[str, Any]:
```

Comment:
Thanks for working on this! I think there's a few UX changes I'd like to see to make this better for Python users:

 1. Combine the first two arguments so the user can always specify just the first argument, rather than sometimes having to go `table.restore(datetime_to_restore="2020-05-01T00:47:31-07:00")`.
 2. Support passing in a `datetime` instead of just a string. That way users don't have to guess which format they need to specify it in.
 3. Make the other arguments keyword-only. It's all boolean arguments and those are terrible without labels (`table.restore(1, True, True)` 😱 )

```suggestion
    def restore(
        self,
        target: Optional[int | datetime | str] = None,
        *,
        ignore_missing_files: bool = False,
        protocol_downgrade_allowed: bool = False,
    ) -> Dict[str, Any]:
```

Reply:
Unfortunately find a problem when I do local test. This caused by a copy action on delta table, the last_modified timestamp change and [we mapping version with timestamp by the file meta](https://github.com/delta-io/delta-rs/blob/main/rust/src/delta.rs#L437). Is this expected behavior? @wjones127 Maybe we should mapping the version with timestamp in commit info.

Reply:
I think that's a hack to make sure the timestamps are monotonically increasing. We could switch to doing what Spark does, but that requires some complicated logic to ensure the parsed timestamps are indeed monotonic. But it would probably work better with the copy action I guess.

Reply:
There has a sphinx doc build step failed in checks. I'm not sure how to fix it. @wjones127 
Here is the error message.
Warning, treated as error:
/home/runner/work/delta-rs/delta-rs/python/deltalake/table.py:docstring of deltalake.table.DeltaTable.restore::py:class reference target not found: datetime.datetime
make: *** [Makefile:87: build-documentation] Error 2
Error: Process completed with exit code 2.

Reply:
Hi @loleek sorry for the delay. I've created a PR into your branch with the fixes for that CI failure: https://github.com/loleek/delta-rs/pull/1

Reply:
Thx.

Reply:
All request changed resolved. Would you mind do a re-review? @wjones127